src/ui/Distance/DistanceResources.ts

```diff
@@ -0,0 +1,245 @@
+     *
+     * If you want to have Kilometers, you should set 1000 as the value.
+     * If you want to have your distance in miles, you should set 1610 as the value, since a mile is approximately 1610 meters.
+     *
+     * The default value is `1000`.
+     */
+    unitConversionFactor: ComponentOptions.buildNumberOption({
+      defaultValue: 1000,
+      validator: value => {
+        return !!value && value > 0;
+      }
+    }),
+    /**
+     * The CSS class for components that needs to be reenabled when the distance is provided.
+     * 
+     * The default value is `coveo-distance-disabled`.
+     */
+    disabledDistanceCssClass: ComponentOptions.buildStringOption({
+      defaultValue: 'coveo-distance-disabled'
+    })
+  };
+
+  /**
+   * Creates a new `DistanceResources` component.
+   * @param element The HTMLElement on which to instantiate the component.
+   * @param options The options for the `DistanceResources` component.
+   * @param bindings The bindings that the component requires to function normally. If not set, these will be
+   * automatically resolved (with a slower execution time).
+   */
+  constructor(public element: HTMLElement, public options: IDistanceOptions, public bindings: IComponentBindings) {
+    super(element, DistanceResources.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, DistanceResources, options);
+    this.bind.onRootElement(InitializationEvents.afterComponentsInitialization, () => this.onAfterComponentsInitialization());
+  }
+
+  /**
+   * Override the current position with the provided values.
+   *
+   * Triggers a query automatically.
+   * @param latitude The latitude to set.
+   * @param longitude The longitude to set.
+   */
+  public setPosition(latitude: number, longitude: number): void {
+    this.latitude = latitude;
+    this.longitude = longitude;
+
+    const args: IPositionResolvedEventArgs = {
+      position: {
+        lat: latitude,
+        long: longitude
+      }
+    };
+
+    this.bind.trigger(this.element, DistanceEvents.onPositionResolved, args);
+    this.registerDistanceQuery();
+    this.queryController.executeQuery();
+  }
+
+  /**
+   * Returns a promise of the last position resolved using the registered position providers.
+   * 
+   * @returns {Promise<IPosition | void>} Promise for the last resolved position value.
+   */
+  public getLastPositionRequest(): Promise<IPosition | void> {
+    return this.lastPositionRequest || Promise.reject('No position request was executed yet.');
+  }
+
+  private onAfterComponentsInitialization(): void {
+    const args: IResolvingPositionEventArgs = {
+      providers: []
+    };
+
+    this.bind.trigger(this.element, DistanceEvents.onResolvingPosition, args);
+
+    this.lastPositionRequest = this.tryGetPositionFromProviders(args.providers)
+      .then(position => {
+        if (!!position) {
+          this.setPosition(position.lat, position.long);
```

Comment:
So after the components are initialized, this component will request a position, and then do a query.

This means that if `autoTriggerQuery` is true on the search interface (which is true most of the time), we end up with multiple queries on initialization.

Reply:
Cannot do much better, if the position request takes a long time, it would block the search interface, which I don't like.

I will at least add an option to disable this query on new position, so that only the next user action would contain the position.

The user still could disable `autoTriggerQuery` and enable this new option, I guess this is the most flexibility we can offer, unless you have a better idea :)